/home/buildnode/jenkins/workspace/swift-PR-Linux@2/branch-master/swift/stdlib/private/SwiftPrivateThreadExtras/ThreadBarriers.swift:84:90: error: use of undeclared type '_stdlib_pthread_barrier_t'

My bad, that has been renamed to `_stdlib_thread_barrier_t`

--- a/stdlib/private/SwiftPrivateThreadExtras/ThreadBarriers.swift
+++ b/stdlib/private/SwiftPrivateThreadExtras/ThreadBarriers.swift
@@ -82,10 +82,10 @@ public func _stdlib_thread_barrier_init(
 }
 
 private func _stdlib_thread_barrier_mutex_and_cond_init(_ barrier: UnsafeMutablePointer<_stdlib_thread_barrier_t>) -> CInt {
-  guard pthread_mutex_init(barrier.pointee.mutex!) == 0 else {
+  guard pthread_mutex_init(barrier.pointee.mutex!, nil) == 0 else {
     return -1
   }
-  guard pthread_cond_init(barrier.pointee.cond!) == 0 else {
+  guard pthread_cond_init(barrier.pointee.cond!, nil) == 0 else {
     pthread_mutex_destroy(barrier.pointee.mutex!)
     return -1
   }

--- a/stdlib/private/SwiftPrivateThreadExtras/ThreadBarriers.swift
+++ b/stdlib/private/SwiftPrivateThreadExtras/ThreadBarriers.swift
@@ -70,7 +70,7 @@ public func _stdlib_thread_barrier_init(
 #else
   barrier.pointee.mutex = UnsafeMutablePointer.allocate(capacity: 1)
   barrier.pointee.cond = UnsafeMutablePointer.allocate(capacity: 1)
-  guard _stdlib_pthread_barrier_mutex_and_cond_init(barrier) == 0 else {
+  guard _stdlib_thread_barrier_mutex_and_cond_init(barrier) == 0 else {
     barrier.pointee.mutex!.deinitialize(count: 1)
     barrier.pointee.mutex!.deallocate()
     barrier.pointee.cond!.deinitialize(count: 1)
@@ -81,7 +81,7 @@ public func _stdlib_thread_barrier_init(
   return 0
 }
 
-private func _stdlib_pthread_barrier_mutex_and_cond_init(_ barrier: UnsafeMutablePointer<_stdlib_pthread_barrier_t>) -> CInt {
+private func _stdlib_thread_barrier_mutex_and_cond_init(_ barrier: UnsafeMutablePointer<_stdlib_thread_barrier_t>) -> CInt {
   guard pthread_mutex_init(barrier.pointee.mutex!) == 0 else {
     return -1
   }

--- a/stdlib/private/SwiftPrivateThreadExtras/ThreadBarriers.swift
+++ b/stdlib/private/SwiftPrivateThreadExtras/ThreadBarriers.swift
@@ -75,6 +75,7 @@ public func _stdlib_thread_barrier_init(
     barrier.pointee.mutex!.deallocate()
     barrier.pointee.cond!.deinitialize(count: 1)
     barrier.pointee.cond!.deallocate()
+    return -1
   }
 #endif
   barrier.pointee.count = count

